parser/src/lib.rs

```diff
@@ -169,8 +169,8 @@ mod test {
 constant %N = 16;
 namespace Fibonacci(%N);
 constant %last_row = (%N - 1);
-let bool = [|X| (X * (1 - X))][0];
-let one_hot = |i, which| match i { which => 1, _ => 0, };
+let bool = [(|X| (X * (1 - X)))][0];
+let one_hot = (|i, which| match i { which => 1, _ => 0, });
```

Comment:
this looks uglier now :D

why are the parens needed? Shouldn't that be done by the parent of that node if needed?

Reply:
It's the same as requiring parens for `a + b`, we add them because the Display trait has no sense of context.
We should fix that at some point and build a prettier expression printer.

Reply:
If you put the responsibility on the parent, you will get `a + b` being formatted as `(1) + (b)`.